File: 백준/Silver/9020. 골드바흐의 추측/골드바흐의 추측.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int isPrime(int n) {
	int prime = 1;
	for (int i = 2; i < (int)sqrt(n) + 1; i++) {
		if (n % i == 0) {
			prime = 0;
			break;
		}
	}
	if (prime) return 1;
	else return 0;
}


int main(void) {
	int n, m;
	scanf("%d", &n); 
	
	while (n--) {
		scanf("%d", &m);
		int num = 2;
		for (int i = 2; i <= m / 2; i++) {
			if (isPrime(i) && isPrime(m - i)) {
				if (i > num) num = i;
			}
		}
		printf("%d %d\n", num, m - num);
	}

	return 0;
}

File: 백준/Silver/9020. 골드바흐의 추측/골드바흐의 추측.py
import math

def isPrime(n):
    if(n < 2):
        return False
    for i in range(2, int(math.sqrt(n) + 1)):
        if(n % i == 0):
            return False
    return True

n = int(input())
while(n > 0):
    m = int(input())
    a, b = m // 2, m // 2
    while(a > 0):
        if(isPrime(a) and isPrime(b)):
            print(a, b)
            break
        else:
            a -= 1
            b += 1
    n -= 1